.github/workflows/snapshot-automation.yml

```diff
@@ -1,11 +1,206 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
 name: Snapshot Release Automation
 on:
   schedule: # schedule the job to run at 12 a.m. daily
     - cron: "0 0 * * *"
 
```

Comment:
Do we need to allow manual triggering of builds and specifying of git commit id to help us perform builds and tests on any commit at any time?

See: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_dispatch

Reply:
Good idea, I will support it soon

Reply:
Manual trigger runs via branch or commitId are now supported